ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -140,9 +320,21 @@ public int compareTo(StoreKey o) {
 
     int result = version.compareTo(other.version);
     if (result == 0) {
```

Comment:
Have changed to use a `switch`, for V1 we just use the old method (keep the old code), and for V2 we compare all the fields. This will ensure that comparison among all existing blobids will not change.